candidate.py

```diff
@@ -211,16 +373,31 @@ def make_candidate(noisy_data):
   try:
     name = normalize_name(data["name"])
     data["name"] = name
-    party = data["party"]
     office = data["office"]
-  except KeyError:
-    return None
+  except KeyError, ex:
+    raise CandidateException("missing expected fields: %s" % ex)
 
-  # Reduce the noise from this very, very noisy dataset.
-  if party and party.lower() != "dem":
-    return None
-  if office and office.lower() == "president":
-    return None
+
+  if office == "house":
+    try:
+      district = data["district"]
+    except KeyError:
+      raise CandidateException("missing expected field: district")
+
+    state, ordinal = normalize_district(district)
+    if state:
+      data['state'] = state
+
+    if ordinal:
+      data['district'] = ordinal
+    else:
+      raise CandidateException("couldn't parse district: %s")
+
+  try:
+    state = normalize_state(data["state"])
```

Comment:
Maybe this belongs in normalize_district? I would have expected that function to do all normalizing, rather than needing two steps.